src/interaction-handlers/games/rps.ts

```diff
@@ -0,0 +1,85 @@
+import {
+  InteractionHandler,
+  InteractionHandlerTypes,
+  Maybe,
+  PieceContext,
+} from '@sapphire/framework';
+import { ButtonInteraction, CommandInteraction, Message, MessagePayload } from 'discord.js';
+import {
+  getCodeyRpsSign,
+  getEmojiFromSign,
+  RpsGame,
+  RpsGameSign,
+  RpsGameStatus,
+  rpsGameTracker,
+} from '../../components/games/rps';
+
+export class RpsHandler extends InteractionHandler {
+  public constructor(ctx: PieceContext, options: InteractionHandler.Options) {
+    super(ctx, {
+      ...options,
+      interactionHandlerType: InteractionHandlerTypes.Button,
+    });
+  }
+
+  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+  // Get the game info and the interaction type
+  public override parse(interaction: ButtonInteraction): Maybe<{
+    gameId: number;
+    sign: RpsGameSign;
+  }> {
+    if (!interaction.customId.startsWith('rps')) return this.none();
+    const parsedCustomId = interaction.customId.split('-');
+    const sign = parsedCustomId[1];
+    const gameId = parseInt(parsedCustomId[2]);
+
+    let gameSign: RpsGameSign;
+    switch (sign) {
+      case 'rock':
+        gameSign = RpsGameSign.Rock;
+        break;
+      case 'paper':
+        gameSign = RpsGameSign.Paper;
+        break;
+      case 'scissors':
+        gameSign = RpsGameSign.Scissors;
+        break;
+      default:
+        gameSign = RpsGameSign.Pending;
+        break;
+    }
+    return this.some({
+      gameId: gameId,
+      sign: gameSign,
+    });
+  }
+
+  public async run(
+    interaction: ButtonInteraction,
+    result: { gameId: number; sign: RpsGameSign },
+  ): Promise<void> {
+    // Don't do anything if the interaction did not come from the player who started the game
+    if (interaction.user.id !== rpsGameTracker.getGameFromId(result.gameId)!.state.player1Id) {
+      return;
+    }
+    rpsGameTracker.runFuncOnGame(result.gameId, (game) => {
+      game.state.player1Sign = result.sign;
+      // If single player, get Codey's sign
+      if (!game.state.player2Id) {
+        game.state.player2Sign = getCodeyRpsSign();
+        game.setStatus(undefined);
+      }
+      if (game.gameMessage instanceof Message) {
+        game.gameMessage.edit(<MessagePayload>game.getGameResponse());
+      } else if (game.gameMessage instanceof CommandInteraction) {
+        game.gameMessage.editReply(<MessagePayload>game.getGameResponse());
+      }
+    });
+    rpsGameTracker.endGame(result.gameId);
+    // We need to do this to avoid "failed interaction" after selecting our option
+    await interaction.reply({
+      content: `You selected ${getEmojiFromSign(result.sign)}.`,
+      ephemeral: true,
+    });
```

Comment:
That's odd; I tried removing the interaction reply and "failed interaction" did not show up for me.

Reply:
Weird, it does not show up for me either now. I'll remove this line